source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionSale/hooks/useCalculateNetProceeds.tsx

```diff
@@ -0,0 +1,79 @@
+import { getIn, useFormikContext } from 'formik';
+import { useEffect } from 'react';
+
+import { DispositionSaleFormModel } from '@/features/mapSideBar/disposition/models/DispositionSaleFormModel';
+import { SystemConstants, useSystemConstants } from '@/store/slices/systemConstants';
+
+export const useCalculateNetProceeds = (isGstEligible: boolean) => {
+  const { values, touched, setFieldValue, isSubmitting } =
+    useFormikContext<DispositionSaleFormModel>();
+  const { getSystemConstant } = useSystemConstants();
+  const gstConstant = getSystemConstant(SystemConstants.GST);
+  const gstDecimal = gstConstant !== undefined ? parseFloat(gstConstant.value) / 100 : undefined;
+
+  const finalSaleAmountTouched = getIn(touched, 'finalSaleAmount');
+  const realtorCommissionTouched = getIn(touched, 'realtorCommissionAmount');
+  const totalCostOfSaleTouched = getIn(touched, 'totalCostOfSale');
+  const netBookAmountTouched = getIn(touched, 'netBookAmount');
+  const gstTouched = getIn(touched, 'gstCollectedAmount');
+  const sppTouched = getIn(touched, 'sppAmount');
+
+  const finalSaleAmount = getIn(values, 'finalSaleAmount');
+  const gstAmount = getIn(values, 'gstCollectedAmount');
+  const realtorCommissionAmount = getIn(values, 'realtorCommissionAmount');
+  const totalCostOfSale = getIn(values, 'totalCostAmount');
+  const netBookAmount = getIn(values, 'netBookAmount');
+  const sppAmount = getIn(values, 'sppAmount');
+
+  useEffect(() => {
+    if (
+      (finalSaleAmountTouched ||
```

Comment:
I'm still not a big fan of these touched checks. If I'm in edit mode, don't I want to see what my current sums are regardless of whether I've touched one of the fields?

Reply:
Regardless of the above, things don't seem to be working correctly. I see this when I first go to a completely filled out sales edit form:

![image](https://github.com/bcgov/PSP/assets/41091511/5aaa19eb-6fb8-47fd-bb40-fe83b0ea0324)

Reply:
then, if I change the final sale price it clears the net proceeds before spp cost field:

![image](https://github.com/bcgov/PSP/assets/41091511/342ad5a6-68bc-4a5d-9770-790a1de1a2de)

Reply:
removed the "touch" from the useEffect, but I am not seing the empty fields as you mention.

Reply:
I'll re-test.